### PR TITLE
Check access to secrets from namespaced brokers

### PR DESCRIPTION
**Pull request by jberkhahn**
per #2508

### CODE REVIEW

File: plugin/pkg/admission/broker/authsarcheck/admission.go [hunk 1]
```diff
@@ -71,57 +71,111 @@ func (s *sarcheck) Admit(a admission.Attributes) error {
...
+		if secretRef == nil {
+			return nil
+		}
+		klog.V(5).Infof("ServiceBroker %+v: evaluating auth secret ref, with authInfo %q", serviceBroker, secretRef)
+		userInfo := a.GetUserInfo()
+
+		sar := &authorizationapi.SubjectAccessReview{
+			Spec: authorizationapi.SubjectAccessReviewSpec{
+				ResourceAttributes: &authorizationapi.ResourceAttributes{
+					Namespace: serviceBroker.Namespace,
```
Comment (MHBauer) [inline plugin/pkg/admission/broker/authsarcheck/admission.go]: this seems to be the only servicebroker specific thing. can we save it off into a namespace variable in the earlier part?

File: plugin/pkg/admission/broker/authsarcheck/admission_test.go [hunk 2]
```diff
@@ -244,4 +244,178 @@ func TestAdmissionBroker(t *testing.T) {
 			t.Errorf("Create test '%s' reports: Unexpected error returned from admission handler: %v", tc.name, err)
 		}
 	}
+
+	// Anonymous struct fields:
+	// name: short description of the testing
+	// broker: a fake broker object
+	// allowed: flag for whether or not the broker should be admitted
```
Comment (MHBauer) [inline plugin/pkg/admission/broker/authsarcheck/admission_test.go]: missing userInfo in comment

File: plugin/pkg/admission/broker/authsarcheck/admission_test.go [hunk 3]
```diff
@@ -244,4 +244,178 @@ func TestAdmissionBroker(t *testing.T) {
...
+				},
+			},
+			userInfo: &user.DefaultInfo{
+				Name:   "system:serviceaccount:test-ns:catalog",
+				Groups: []string{"system:serviceaccount", "system:serviceaccounts:test-ns"},
+			},
+			allowed: true,
+		},
+		{
+			name: "namespace broker with bearer token, unauthenticated user",
```
Comment (MHBauer) [inline plugin/pkg/admission/broker/authsarcheck/admission_test.go]: what's the significants of bearer token and unauthenticated together? Is there an unauthenticated on it's own, I think it's just copy and paste from above and the cases don't necessarily make sense.

Comment (jberkhahn) [reply plugin/pkg/admission/broker/authsarcheck/admission_test.go]: the bearer token is platform <-> broker, not user <-> platform

**Comment by jboyd01** [comment]
Thanks for fixing this & addition of new test cases.
/approve

**Comment by jboyd01** [comment]
adding lgtm, mhb had added but it doesn't seem prow caught it.
/lgtm

**Comment by MHBauer** [comment]
maybe you can't lgtm a hold?

**Pull request merged**